assets/js/editor-components/default-notice/index.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { store as editorStore } from '@wordpress/editor';
+import triggerFetch from '@wordpress/api-fetch';
+import { store as coreStore } from '@wordpress/core-data';
+import { Notice, Button } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { CHECKOUT_PAGE_ID, CART_PAGE_ID } from '@woocommerce/block-settings';
+import { useCallback, useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+export function DefaultNotice( { block }: { block: string } ) {
+	// To avoid having the same logic twice, we're going to handle both pages here.
+	const ORIGINAL_PAGE_ID =
+		block === 'checkout' ? CHECKOUT_PAGE_ID : CART_PAGE_ID;
+	const settingName =
+		block === 'checkout'
+			? 'woocommerce_checkout_page_id'
+			: 'woocommerce_cart_page_id';
+
+	const noticeContent =
+		block === 'checkout'
+			? __(
+					'If you would like to use this block as your default checkout, update your page settings',
+					'woo-gutenberg-products-block'
+			  )
+			: __(
+					'If you would like to use this block as your default cart, update your page settings',
+					'woo-gutenberg-products-block'
+			  );
```

Comment:
We handle Cart and Checkout from this component, everything is the same we just need to have 2 different values.